components/ui/CopyButton.tsx

```diff
@@ -0,0 +1,48 @@
+import { ClipboardIcon, CheckIcon } from "@heroicons/react/24/solid";
+import { useState } from "react";
+import classNames from "classnames";
+import Tippy from "@tippyjs/react";
+
+interface CopyButtonProps {
+  value: string | number | null;
+}
+
+export const CopyButton = ({ value }: CopyButtonProps) => {
+  const [copied, setCopied] = useState(false);
+
+  const handleCopy = () => {
+    if (value !== null) {
+      navigator.clipboard.writeText(value.toString());
+      setCopied(true);
+      setTimeout(() => setCopied(false), 1000);
+    }
+  };
+
+  return (
+    <Tippy content={copied ? "Copied!" : "Copy to clipboard"} placement="right">
+      <button
+        onClick={handleCopy}
+        className={classNames(
```

Comment:
For all other buttons, we went with a different hover pattern (no padding, just changing reducing the opacity on hover) so I would tend to prefer that for consistency

Reply:
Done: here: 5fcad5be199abb89bbbb253225d314c507741066